src/plugin-recall-storage/services/recall.service.ts

```diff
@@ -287,84 +454,105 @@ export class RecallService extends Service {
         'Get/Create bucket',
       );
 
-      const unsyncedLogs = await this.runtime.databaseAdapter.getUnsyncedLogs();
-      const filteredLogs = unsyncedLogs.filter((log) => log.type === 'chain-of-thought');
+      // Get logs that haven't been synced yet
+      const unsyncedLogs = await this.getUnsyncedLogs();
 
-      if (filteredLogs.length === 0) {
+      if (unsyncedLogs.length === 0) {
         elizaLogger.info('No unsynced logs to process.');
         return;
       }
 
-      elizaLogger.info(`Found ${filteredLogs.length} unsynced logs.`);
+      elizaLogger.info(`Found ${unsyncedLogs.length} unsynced logs.`);
 
       let batch: string[] = [];
       let batchSize = 0;
-      let syncedLogIds: UUID[] = [];
-      let failedLogIds: UUID[] = [];
+      let processedLogIds: string[] = [];
```

Comment:
i ran into some problems when i stop/restart the agent, and there's an unsynced entry in the sqlite db. the `batch` will initially be empty, and we pass this value down below in `this.storeBatchToRecall(bucketAddress, batch, batchTimestamp);`.

the `storeBatchToRecall`'s upload fails due to some low-level entangler issue (i'll look into this in the SDK), but the root cause is the fact the array is empty.

maybe this is just an edge case on stop/restart scenarios tho